### PR TITLE
Port yuzu-emu/yuzu#4577: "common/assert: Make use of C++ attribute syntax"

### PR DESCRIPTION
**Pull request by FearlessTobi**
See yuzu-emu/yuzu#4577 for more details.

**Original description**:
Normalizes the syntax used for attributes

---
This change is [Reviewable](https://reviewable.io/reviews/citra-emu/citra/5529)

### CODE REVIEW

File: src/common/assert.h [hunk 1]
```diff
@@ -17,11 +17,12 @@
 // enough for our purposes.
 template <typename Fn>
 #if defined(_MSC_VER)
-__declspec(noinline, noreturn)
+[[msvc::noinline, noreturn]]
```
Comment (BreadFish64) [inline src/common/assert.h:20]:
```suggestion
[[ msvc::noinline, noreturn ]]
```

File: src/common/assert.h [hunk 2]
```diff
@@ -17,11 +17,12 @@
...
 #elif defined(__GNUC__)
-    __attribute__((noinline, noreturn, cold))
+[[gnu::cold, gnu::noinline, noreturn]]
```
Comment (BreadFish64) [inline src/common/assert.h:22]:
```suggestion
[[ gnu::cold, gnu::noinline, noreturn ]]
```

**Comment by FearlessTobi** [comment]
`clang-format version 6.0.1-14`
Our buildbot is using an outdated clang version. 
That's why it wants those changes.

**Comment by BreadFish64** [comment]
I see

**Review by zhaowenlan1779 — APPROVED** [review]

**Pull request merged**